tsl/test/sql/continuous_aggs_errors.sql

```diff
@@ -480,5 +480,45 @@ SELECT _timescaledb_internal.refresh_continuous_aggregate(
 \set ON_ERROR_STOP 1
 \set VERBOSITY terse
 
-DROP TABLE measurements;
+-- Add a continuous aggregate on the measurements table and a policy.
+CREATE MATERIALIZED VIEW measurements_summary WITH (timescaledb.continuous) AS
+SELECT time_bucket('1 day', time), COUNT(time)
+  FROM measurements
+GROUP BY 1 WITH NO DATA;
+
+-- First test that add_job checks the config. These should all generate errors.
+\set ON_ERROR_STOP 0
+SELECT add_job(
```

Comment:
Why use `add_job` here instead of the job-specific wrappers? Should we test both? At least having an explanation for this would make sense.

I am also wondering if we should actually raise an error for this case and point to the policy-specific APIs when people are adding policies using our "official" policy functions.

Reply:
There is no need to test the job-specific wrappers since they take correctly typed arguments and construct the config. The function `add_job` accepts a configuration and we need to do type-checking here so making sure that it works.

It might be reasonable to not allow creating non-custom jobs using this API, but this is outside the scope of this PR and warrant a separate discussion, so we should put this in a separate PR.

Reply:
I am not sure I agree. This seems like a strange way to test validation in `add_job` simply because custom jobs (which this function is typically used for) currently doesn't do any validation of the config.

So, why test something that is not the way we'd like people to use this? It seems odd, in particular since we currently have no validation of the config for custom jobs, where it really would matter for `add_job`.

I think your comment about job-specific wrappers validating the config in other ways (with typed arguments) says it all: we now have two different ways of adding the same jobs where one way provides must stronger validation through typed arguments. That concerns me.

Reply:
I agree with Erik here, unless we check those cases elsewhere we should also test job wrappers to see that they act in the same or valid manner.

Reply:
The job wrappers are already tested elsewhere: they build the config so there is nothing to check. The only purpose of this test is to check that users cannot inject an invalid config using this interface. I think that any changes to the current API requires a separate discussion, so it is out of scope for this PR.